src/python_testing/TC_DA_1_2.py

```diff
@@ -56,7 +56,7 @@ def parse_single_vidpid_from_common_name(commonName: str, tag_str: str) -> str:
         return None
 
     s = sp[1][:4]
-    if not s.isupper() or len(s) != 4:
+    if not (s.isnumeric() or s.isupper()) or len(s) != 4:
```

Comment:
Overall, the entire if could be replaced with `if re.match("[0-9A-F]{4}", s) is None:`

Reply:
Changed, added new test vector and confirmed in test that this now fails on line 81 at this function call, rather than on line 90 in the int conversion.